### ISSUE TITLE
Next Step Microgrant: Jackrabbit Finance (DEX on FVM)

### ISSUE THREAD

**Issue by Islandersfan2016** [filecoin-project/devgrants#1308]
### 1. What is your project, and what problem does it solve? (max 100 words)
Jackrabbit Finance is the first DEX to be deployed on the FVM (Filecoin Virtual Machine). The DEX was introduced in the EthGlobal FVM Hackathon and is now a part of the FVM Foundry Program. The Exchange will feature cross-chain token swaps, liquidity pools, and decentralized bonds. The DEX will utilize the new EIP 3475 On-chain storage bonds to have a bond market for users to stake, buy, and sell bonds.

### 2. Project links
  
Github: https://github.com/Jackrabbit-Finance
Website: http://www.jackrabbitfinance.xyz/
DEX Demo: Coming Soon! 

  
### 3. a) How is IPFS, Filecoin, or related technology used in this project? (max 200 words)
The DEX will be deployed (native) on the Filecoin Virtual Machine. The decentralized bonds will have coinciding IPFS files containing additional regulatory information. Jackrabbit will be using Bridges move assets to add liquidity to FVM. Also, Storage Providers as a way to raise funds within the Filecoin Ecosystem. Instead of normal DeFi lending and borrowing (AAVE, Compound), the bond tokens offer an easier way for SPs to raise funds without putting up high amounts of collateral.
### 5. Do you agree to share grant reports upon request, including a final grant report at the end of the three month period?
Yes, the plan is to launch the exchange on the day the FVM mainnet is launched. I am prepared to make weekly updates as the project is a part of the FVM foundry program.
  
### 6. Does your proposal comply with our Community Code of Conduct?
Yes

### 7. Links and submissions
no prizes
Submission link: https://ethglobal.com/showcase/jackrabbit-finance-df4dm
  
### Additional questions:
Name: Lamont Veal
Email: Lamont.veal@jrt.finance
Github: islandersfan2016
Role: Lead Developer

* How did you learn about our microgrant program? 
FVM Foundry Program

* If your project was created as part of an event or hackathon:
Hack FEVM
https://ethglobal.com/showcase/jackrabbit-finance-df4dm

**Comment by ErinOCon** [comment]
Hi @Islandersfan2016, thank you for your proposal! Before proceeding with this review, can you direct us to the project results for [KopperDAO](https://github.com/filecoin-project/devgrants/issues/329)?

**Comment by Islandersfan2016** [comment]
Hi Erin,

The Project is Jackrabbit. KopperDao was a different project. I have the
results of Jackrabbit here. We are continuing to build and adding
Documentation as the FVM's Mainnet is not ready yet. I have added the links
to the Jackrabbit page, the Github, and attached the Demo Video and doc
with contract addresses deployed on Wallaby.

http://www.jrt.finance/

Jackrabbit Finance (github.com) <https://github.com/Jackrabbit-Finance>
 Jackrabbit and 2 more pages - Personal - Micros...
<https://drive.google.com/file/d/19pKOeBHZVNN9luxEG9jg6iNO_PFiV5B-/view?usp=drive_web>

**Comment by Islandersfan2016** [comment]
Hi Erin,

The KopperDAO project was renamed Muse Filez. This project team (different
team than Jackrabbit) is still continuing to build out Muse Filez. We are
creating our own NFT contracts for access control to the IPFS files (before
we were using the Unlock Protocol). We will be debuting and deploying these
contracts in the EthGlobal FVM hackathon.

When the hackathon is finished we will send you the demo video with links.

MuseFilez (github.com) <https://github.com/MuseFilez>

Muse App (mfilez.io) <https://www.mfilez.io/>

Thank you,
Lamont V.

**Comment by ErinOCon** [comment]
Hi @Islandersfan2016, thank you for your response. To avoid confusion, I have sent a follow up question that applies for your previous project. Once this information is provided, we can proceed with our review.

**Comment by Islandersfan2016** [comment]
Hi @ErinOCon I have updated out Grant's proposal. We will use this grant to pay for our smart contract audit. Plus, you can follow our progress on the development with the updated links I've added as well.

**Comment by eshon** [comment]
Hi @Islandersfan2016, 2 questions:
- Other than deploying existing EVM DeFi smart contracts to FVM, what new contributions do you plan to add to this project that are specific to Filecoin and Filecoin's VM?
- Have you received any audit quotes that can be covered by this grant?

**Comment by Islandersfan2016** [comment]
Hi, Jackrabbit is a Decentralized Bond Exchange. We are utilizing the EIP 3475 (Abstract Storage Bonds) standard. These bond tokens allow for bond issuers to interact directly with the bond investors and also sell their bond tokens on a secondary marketplace. Our liquidity pools back our bond tokens to minimize the risk for investors. For example, a bond token redeemed in FIL will be backed by a liquidity containing FIL tokens if the bond issuer is not able to make a full payment to the investor at redemption. 

Yes, we have forked Uniswap's V2 contracts (as do most new DEXs), BUT we have not simply deployed existing smart contracts onto the FVM since ERC 3475's contracts are not widely used at all within DeFi. Plus, any forked contracts have been modified to be unique to Jackrabbit's protocol.

Also, the bonds are for Storage Providers as a way to raise funds within the Filecoin Ecosystem. Instead of normal DeFi lending and borrowing (AAVE, Compound), the bond tokens offer an easier way for SPs to raise funds without putting up high amounts of collateral. 

We have not received a quote from a smart contract auditor yet. We've messaged Consensys Diligence and are waiting for a reply. We have received a quote from an attorney on her retainer. 

We will be moving this application to the Open Grants. As we come close to launching a testnet version of the entire DEX on Hyperspace. The team will need the smart contract audit and an attorney on retainer to launch to main net. I will be updating this when we have a definitive quote on both (attorney, audit).
